crates/rpc/rpc/src/engine/jwt_secret.rs

```diff
@@ -45,23 +45,38 @@ const JWT_SIGNATURE_ALGO: Algorithm = Algorithm::HS256;
 /// [Secret key - Engine API specs](https://github.com/ethereum/execution-apis/blob/main/src/engine/authentication.md#key-distribution)
 #[derive(Clone)]
 #[allow(missing_debug_implementations)]
-pub struct JwtSecret(String);
+pub struct JwtSecret([u8; 32]);
+
+impl TryInto<JwtSecret> for String {
+    type Error = JwtError;
 
-impl JwtSecret {
     /// Creates an instance of [`JwtSecret`][crate::engine::JwtSecret].
     /// Generates and error if one of the following applies:
     /// - The `hex` argument is not a valid hexadecimal string
     /// - The `hex` argument length is less than `JWT_SECRET_LEN`
-    pub fn new(hex: String) -> Result<JwtSecret, JwtError> {
-        let hex: String = hex.trim().into();
+    fn try_into(self) -> Result<JwtSecret, Self::Error> {
+        let hex: String = self.trim().into();
         if hex.len() != JWT_SECRET_LEN {
             Err(JwtError::InvalidLength(JWT_SECRET_LEN, hex.len()))
         } else {
-            hex::decode(&hex)?;
-            Ok(Self(hex))
+            let hex_bytes = hex::decode(&hex)?;
+            let mut bytes = [0; 32];
+            let mut writer = &mut bytes[..];
+            let _ = writer.write(hex_bytes.get(0..32).expect("This should never happen"));
+            Ok(JwtSecret(bytes))
```

Comment:
can use `bytes.as_slice().try_into()` here directly

Reply:
You refer to the instantiation of `JwtSecret`? If so I removed the `TryInto` impl, so it's implicitly solved.
Otherwise can you be more clear?